test/common/entity/get_states.test.ts

```diff
@@ -0,0 +1,499 @@
+import type { HassEntity } from "home-assistant-js-websocket";
+import { beforeEach, describe, expect, it } from "vitest";
+import { getStates } from "../../../src/common/entity/get_states";
+import type { HomeAssistant } from "../../../src/types";
+
+describe("getStates", () => {
+  let mockHass: HomeAssistant;
+
+  beforeEach(() => {
+    mockHass = {
+      states: {
+        "zone.home": {
+          entity_id: "zone.home",
+          state: "zoning",
+          attributes: {
+            friendly_name: "Home",
+          },
+        } as HassEntity,
+        "zone.work": {
+          entity_id: "zone.work",
+          state: "zoning",
+          attributes: {
+            friendly_name: "Work",
+          },
+        } as HassEntity,
+        "zone.office": {
+          entity_id: "zone.office",
+          state: "zoning",
+          attributes: {
+            friendly_name: "Office",
+          },
+        } as HassEntity,
+      },
+      locale: {
+        language: "en",
+      },
+    } as any;
+  });
+
+  const createMockEntity = (
+    entity_id: string,
+    state = "on",
+    attributes: Record<string, any> = {}
+  ): HassEntity => ({
+    entity_id,
+    state,
+    attributes,
+    context: { id: "test", parent_id: null, user_id: null },
+    last_changed: "2023-01-01T00:00:00Z",
+    last_updated: "2023-01-01T00:00:00Z",
+  });
+
+  describe("Fixed domain states", () => {
+    it("should return alarm control panel states", () => {
+      const entity = createMockEntity("alarm_control_panel.test");
+      const result = getStates(mockHass, entity);
+
+      expect(result).toContain("armed_away");
+      expect(result).toContain("armed_home");
+      expect(result).toContain("disarmed");
+      expect(result).toContain("triggered");
+      expect(result).toContain("unavailable");
+      expect(result).toContain("unknown");
```

Comment:
We should check the `length` on all these results, so they fail when a new unexpected state is added.